pkg/endpoint/providers/kube/client.go

```diff
@@ -25,13 +24,13 @@ func NewProvider(kubeClient kubernetes.Interface, kubeController k8s.Controller,
 	informerFactory := informers.NewSharedInformerFactory(kubeClient, k8s.DefaultKubeEventResyncInterval)
 
 	informerCollection := InformerCollection{
-		Endpoints:   informerFactory.Core().V1().Endpoints().Informer(),
-		Deployments: informerFactory.Apps().V1().Deployments().Informer(),
+		Endpoints: informerFactory.Core().V1().Endpoints().Informer(),
+		Pods:      informerFactory.Core().V1().Pods().Informer(),
```

Comment:
We will probably want to move this to KubeController, as `xds_certificate.go` (see `listServicesForPod` et al) needs to do the exact same when looking up a pod/service relation based on cert. 

Not blocking this though, this is a good change.